fabtests/pytest/common.py

```diff
@@ -223,18 +291,18 @@ def __init__(self, cmdline_args, executable,
 
         self._cmdline_args = cmdline_args
         self._timeout = timeout or cmdline_args.timeout
-        self._server_base_command = self.prepare_base_command("server", executable, iteration_type,
+        self._server_base_command, server_additonal_environment = self.prepare_base_command("server", executable, iteration_type,
                                                               completion_type, prefix_type,
                                                               datacheck_type, message_size,
                                                               memory_type, warmup_iteration_type)
-        self._client_base_command = self.prepare_base_command("client", executable, iteration_type,
+        self._client_base_command, client_additonal_environment = self.prepare_base_command("client", executable, iteration_type,
                                                               completion_type, prefix_type,
                                                               datacheck_type, message_size,
                                                               memory_type, warmup_iteration_type)
 
 
-        self._server_command = self._cmdline_args.populate_command(self._server_base_command, "server", self._timeout)
-        self._client_command = self._cmdline_args.populate_command(self._client_base_command, "client", self._timeout)
+        self._server_command = self._cmdline_args.populate_command(self._server_base_command, "server", self._timeout, server_additonal_environment)
+        self._client_command = self._cmdline_args.populate_command(self._client_base_command, "client", self._timeout, client_additonal_environment)
 
     def prepare_base_command(self, command_type, executable,
```

Comment:
Please check the return branches in this method. There are a few missed cases that only return 1 value.

Reply:
added 1. Did not find more.